_projects/2024/100452420/100452420.Rmd

```diff
@@ -0,0 +1,364 @@
+
+
+```{r, layout = "1-body-outset", fig.width= 10, fig.height= 3.8} 
+
+p <- ggplot(serena_data, aes(x = ranking_date, y = rank)) +
+  geom_line(color = serena_data$line_color,
+            size = ifelse(serena_data$line_color == "#FF851B", 1.5, 1)) +
+  scale_y_reverse(limits = c(20, 1),
+                  breaks = c(1, 5, 10, 15, 20),
+                  labels = c("No. 1" = expression(bold("No. 1")), "5", "10", "15", "20")) +
+  scale_x_date(
+    limits = c(as.Date("1998-01-01"), as.Date("2022-12-31")),
+    breaks = seq(as.Date("1998-01-01"), as.Date("2022-01-01"), by = "1 year"),
+    date_labels = "          |  '%y"
+  )
+p
+```
+
+
+To match the style of the original chart, I used theme_minimal and customized grid lines and axis labels.
+
+```{r, layout = "1-body-outset", fig.width= 10, fig.height= 3.8} 
+
+p <- p +
+  theme_minimal(base_size = 10) +
+  theme(
+    axis.text.x = element_text(color = "grey"),
+    plot.title = element_text(hjust = 0.47, face = "bold", size = 10),
+    axis.title.y = element_blank(),
+    axis.title.x = element_blank(),
+    axis.text.y = element_text(hjust = 1, color = "black"),
+    panel.grid.major.x = element_blank(),
+    panel.grid.minor = element_blank())
+p
+```
+
+### Annotations and Titles
+
+Annotations provide context to key moments in Serena's career. I included the four comments from the original graph. Additionally, I added the title to the plot. Here's how I implemented these changes:
+
+
+```{r, layout = "1-body-outset", fig.width= 10, fig.height= 3.8} 
+p <- p +
+  labs(
+    title = "Serena Williams's world ranking since 1998",
+    x = "",
+    y = ""
+  ) +
+  annotate("text", x = as.Date("2001-10-01"), y = 1.8,
+           label = "Williams first rose to\nNo. 1 in July 2002, at\nage 20. She stayed\nNo. 1 for 49 weeks.",
+           size = 2.5, hjust = 1) +
+  annotate("text", x = as.Date("2005-02-01"), y = 18,
+           label = "Her ranking\nfell below\n20 in 2006.",
+           size = 2.5, hjust = 1) +
+  annotate("text", x = as.Date("2013-08-01"), y = 3.2,
+           label = "She stayed No.1 for\n186 consecutive weeks\nfrom Feb. 2013 to\nSept. 2016.",
+           size = 2.5, hjust = 0) +
+  annotate("text", x = as.Date("2017-02-01"), y = 18,
+           label = "She gave birth to\nher daughter in\n2017 and returned\nto tennis in 2018.",
+           size = 2.5, hjust = 1)
+p
+
+```
+
+### Last Adjustments
+
+Finally, the arrows, dotted lines, and source were added to the chart. Both the arrows and dotted lines were created using the annotate function. The source is placed at the bottom of the chart, as in the original. Although I did not obtain the data directly from the WTA, I aimed to replicate the original plot as closely as possible.
+
+
+```{r, layout = "1-body-outset", fig.width= 10, fig.height= 3.9} 
```

Comment:
Please add `preview = TRUE` so that this image is shown in the list of projects.